File: like-service/src/main/java/com/backend/like/model/PostLike.java
package com.backend.like.model;

import java.time.Instant;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "t_post_like", uniqueConstraints = @UniqueConstraint(columnNames = { "post_uuid", "user_uuid" }))
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class PostLike {

  @Column(name = "id", nullable = false, updatable = false, unique = true)
  @NotNull(message = "id is required")
  @Id
  @GeneratedValue(strategy = GenerationType.AUTO, generator = "like_sequence")
  @SequenceGenerator(name = "like_sequence", sequenceName = "like_sequence", allocationSize = 10)
  private Long id;

  @Column(name = "post_uuid", nullable = false, updatable = false, unique = true, columnDefinition = "CHAR(36)")
  @NotNull(message = "uuid is required")
  private String postUUID;

  @Column(name = "user_uuid", nullable = false, updatable = false, unique = true, columnDefinition = "CHAR(36)")
  @NotNull(message = "uuid is required")
  private String userUUID;

  @Column(name = "liked_at_timestamp", nullable = false, updatable = false)
  @Builder.Default
  private Instant likedAt = Instant.now();

}

File: like-service/src/main/kotlin/com/backend/like/model/PostLike.kt
package com.backend.like.model

import jakarta.persistence.*
import java.time.Instant

@Entity
@Table(name = "post_likes")
class PostLike(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(name = "post_uuid", nullable = false)
    val postUUID: String,

    @Column(name = "user_uuid", nullable = false)
    val userUUID: String,


    @Column(name = "liked_at_timestamp", nullable = false, updatable = false)
    val likedAt: Instant = Instant.now(),

)